src/generate.dart

```diff
@@ -0,0 +1,283 @@
+import 'dart:async';
+import 'dart:io';
+import 'dart:convert';
+
+import 'package:args/args.dart';
+import 'package:path/path.dart' as path;
+
+/// The type of the process runner callback.  This allows us to
+/// inject a fake process runner into the DiagramGenerator for tests.
+typedef Future<ProcessResult> ProcessRunner(String executable, List<String> arguments,
+    {String workingDirectory,
+    Map<String, String> environment,
+    bool includeParentEnvironment,
+    bool runInShell,
+    Encoding stdoutEncoding,
+    Encoding stderrEncoding});
```

Comment:
i think if you add a trailing comma here you may get slightly cleaner results from dartfmt. The above is pretty terrible.

Reply:
ditto various others below

Reply:
I did a pass fixing up formatting now that I know a bit more about formatting Flutter code.